### ISSUE TITLE
`$XDG_CACHE_HOME` is partially ignored

### ISSUE THREAD

**Issue by MartinX3** [nicolargo/glances#1802]
**Describe the bug**
It seems to be partially ignored `$XDG_CACHE_HOME` and uses `$HOME/.cache` instead.
`Log file: /tmp/.cache/martin/glances/glances.log`
But there is a `/home/martin/.cache/glances/glances-version.db` also.

**To Reproduce**
1. Set $XDG_CACHE_HOME in `/etc/profile.d/`

**Expected behavior**
Using $XDG_CACHE_HOME

**Desktop (please complete the following information):**
 - OS: Arch Linux / EndeavourOS
 - Glances Version Glances v3.1.5 with PsUtil v5.8.0
 - Glances logs file [output of tail -10 <glances logs file>]
```
2021-01-24 20:49:18,938 -- INFO -- Start Glances 3.1.5
2021-01-24 20:49:18,938 -- INFO -- CPython 3.9.1 and psutil 5.8.0 detected
```

**Comment by nicolargo** [comment]
Hi @MartinX3 

do not reproduced on my side (Ubuntu 20.04).

Please can you make the following test:

```
$ python
>>> import os
>>> os.environ.get('XDG_CACHE_HOME') or os.path.expanduser('~/.cache')
```

**Comment by MartinX3** [comment]
Hi @nicolargo 

```
$ python
Python 3.9.1 (default, Dec 13 2020, 11:55:53) 
[GCC 10.2.0] on linux
>>> import os
>>> os.environ.get('XDG_CACHE_HOME')
'/tmp/.cache/martin'
>>> os.path.expanduser('~/.cache')
'/home/martin/.cache'
```

But I fixed it now.

I only set `XDG_CACHE_HOME` in `/etc/profile.d/` (readable in `env` and `printenv`)
Then it set the log file correctly but put the .db file into my homedir
(Almost every application reads it from the profile folder.)

After I set `XDG_CACHE_HOME` a again into `/etc/environment` where it is loaded by `pam_env` for systemd (and dbus services via systemd-dbus/sd-bus) it works finally for the log and the .db file.

It's just sad, that I need to declare `XDG_CACHE_HOME` at two different places.
But that's probably some posix/systemd thing.
